src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1568,6 +1568,7 @@ impl Step for MirOpt {
                 run(panic_abort_target);
             }
         } else {
+            // If we're keeping a std stage, only run tests for this target.
             run(self.target);
```

Comment:
My proposed patch would be to simply move this to before the `if`.

Alternatively we could enact a policy saying that `only-x86_64` is not allowed in mir-opt, and generally making sure that the set of targets in the bless list above is covering all mir-opt tests. That would require tool support however.

Reply:
It would be nice to have more targets here. At least have all targets that are run in CI

Reply:
That would take forever though